cmd/operator-sdk/cli/cli.go

```diff
@@ -42,91 +30,71 @@ import (
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+	"sigs.k8s.io/kubebuilder/pkg/cli"
+	kbgov1 "sigs.k8s.io/kubebuilder/pkg/plugin/v1"
+	kbgov2 "sigs.k8s.io/kubebuilder/pkg/plugin/v2"
 )
 
-// GetCLIRoot is intended to creeate the base command structure for the OSDK for use in CLI and documentation
-func GetCLIRoot() *cobra.Command {
-	root := &cobra.Command{
-		Use:   "operator-sdk",
-		Short: "An SDK for building operators with ease",
-		PersistentPreRun: func(cmd *cobra.Command, args []string) {
-			if viper.GetBool(flags.VerboseOpt) {
-				if err := projutil.SetGoVerbose(); err != nil {
-					log.Fatalf("Could not set GOFLAGS: (%v)", err)
-				}
-				log.SetLevel(log.DebugLevel)
-				log.Debug("Debug logging is set")
-			}
-			if err := checkGoModulesForCmd(cmd); err != nil {
-				log.Fatal(err)
-			}
-		},
-	}
+var commands = []*cobra.Command{
+	// Once the KB CLI is made the default, add the "new" command as a way to
+	// scaffold the legacy project layout and mark "new" as deprecated
+	// new.NewCmd()
 
-	root.AddCommand(
-		add.NewCmd(),
-		alpha.NewCmd(),
-		build.NewCmd(),
-		bundle.NewCmd(),
-		cleanup.NewCmd(),
-		completion.NewCmd(),
-		execentrypoint.NewCmd(),
-		generate.NewCmd(),
-		migrate.NewCmd(),
-		new.NewCmd(),
-		olm.NewCmd(),
-		printdeps.NewCmd(),
-		run.NewCmd(),
-		scorecard.NewCmd(),
-		test.NewCmd(),
-		version.NewCmd(),
-	)
+	alpha.NewCmd(),
+	bundle.NewCmd(),
+	cleanup.NewCmd(),
+	completion.NewCmd(),
+	olm.NewCmd(),
+	run.NewCmd(),
+	scorecard.NewCmd(),
+	test.NewCmd(),
+	version.NewCmd(),
+	// TODO(hasbro17): add generate csv command after aligning it for kubebuilder layout
```

Comment:
> We could take advantage and create the testdata dir to scaffold a project to test it as it is done in the kubebuilder

IMO we shouldn't duplicate kubebuilder tests. Once we get to the point of making additional tweaks or changes to the Kubebuilder scaffolds or when we add our own plugins with custom layouts, it might make sense to add similar tests. Until then, I don't see how it adds any value, unless I'm misunderstanding something.

> Also, I do not think that we can merge this PR in the master until plugin phase 1 be merged there as well.

Typically I would agree with this, but I think in this case, it might make sense to push forward on this since:
* We're not making the KB CLI the default yet
* We're just now introducing the KB dependency, so our existing legacy code is not affected by the use of a feature branch commit.

What do others think?

Reply:
HI @joelanford, 

I got your point of view over merge based on the branch and then it is OK for me. 
Regards the tests I really think that we should add them as soon as possible because as you said now we have nothing on top but we will have it. However, IHMO it is not a blocker to merge this PR at all. Would just make easier perform the review.   I am checking it manually anyway.